spring-integration-file/src/main/java/org/springframework/integration/file/remote/RemoteFileTemplate.java

```diff
@@ -87,60 +89,120 @@
 
 	private volatile BeanFactory beanFactory;
 
+	/**
+	 * Construct a {@link RemoteFileTemplate} with the supplied session factory.
+	 * @param sessionFactory the session factory.
+	 */
 	public RemoteFileTemplate(SessionFactory<F> sessionFactory) {
 		Assert.notNull(sessionFactory, "sessionFactory must not be null");
 		this.sessionFactory = sessionFactory;
 	}
 
+	/**
+	 * Determine whether the remote directory should automatically be created when
+	 * sending files to the remote system.
+	 * @param autoCreateDirectory true to create the directory.
+	 */
 	public void setAutoCreateDirectory(boolean autoCreateDirectory) {
 		this.autoCreateDirectory = autoCreateDirectory;
 	}
 
+	/**
+	 * Set the file separator when dealing with remote files; default '/'.
+	 * @param remoteFileSeparator the separator.
+	 */
 	public void setRemoteFileSeparator(String remoteFileSeparator) {
 		Assert.notNull(remoteFileSeparator, "'remoteFileSeparator' must not be null");
 		this.remoteFileSeparator = remoteFileSeparator;
 	}
 
+	/**
+	 * @return the remote file separator.
+	 */
 	public final String getRemoteFileSeparator() {
 		return remoteFileSeparator;
 	}
 
+	/**
+	 * Set the remote directory expression used to determine the remote directory to which
+	 * files will be sent.
+	 * @param remoteDirectoryExpression the remote directory expression.
+	 */
 	public void setRemoteDirectoryExpression(Expression remoteDirectoryExpression) {
 		Assert.notNull(remoteDirectoryExpression, "remoteDirectoryExpression must not be null");
 		this.directoryExpressionProcessor = new ExpressionEvaluatingMessageProcessor<String>(remoteDirectoryExpression, String.class);
 	}
 
+	/**
+	 * Set a temporary remote directory expression; used when transferring files to the remote
+	 * system. After a successful transfer the file is renamed using the
+	 * {@link #setRemoteDirectoryExpression(Expression) remoteDirectoryExpression}.
+	 * @param temporaryRemoteDirectoryExpression the temporary remote directory expression.
+	 */
 	public void setTemporaryRemoteDirectoryExpression(Expression temporaryRemoteDirectoryExpression) {
 		Assert.notNull(temporaryRemoteDirectoryExpression, "temporaryRemoteDirectoryExpression must not be null");
 		this.temporaryDirectoryExpressionProcessor = new ExpressionEvaluatingMessageProcessor<String>(temporaryRemoteDirectoryExpression, String.class);
 	}
 
+	/**
+	 * Set the file name expression to determine the name of the local file when retrieving
+	 * a remote file.
```

Comment:
Something wrong is here:
```
* Set the expression for remote file to retrieve.
```
?
````java
public boolean get(Message<?> message, InputStreamCallback callback) {
	Assert.notNull(this.fileNameProcessor, "A 'fileNameExpression' is needed to use get");
	String remotePath = this.fileNameProcessor.processMessage(message);
	return this.get(remotePath, callback);
}
````
It is exactly a `remote path`. There is nothing about `local`